src/test/java/emissary/core/InterruptibleCharSequenceTest.java

```diff
@@ -0,0 +1,104 @@
+package emissary.core;
+
+import emissary.test.core.junit5.UnitTest;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.regex.Pattern;
+
+import static java.lang.Thread.sleep;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+class InterruptibleCharSequenceTest extends UnitTest {
+    public static String BACKTRACKER = "^(((((a+)*)*)*)*)*$";
+    public static String INPUT = "aaaaab";
+
+    // Uncomment to test overhead. This method does not perform any assertions.
+    // @Test
+    void testOverhead() {
+        String regex = BACKTRACKER;
+        String inputNormal = INPUT;
+        InterruptibleCharSequence inputWrapped = new InterruptibleCharSequence(inputNormal);
+        // warm up
+        tryMatch(inputNormal, regex);
+        tryMatch(inputWrapped, regex);
+        int trials = 5;
+        long stringTotalTime = 0;
+        long interruptibleCharSequenceTotalTime = 0;
+        for (int i = 1; i <= trials; i++) {
+            logger.info("trial {} of {}", i, trials);
+            long start = System.currentTimeMillis();
+            tryMatch(inputNormal, regex);
+            long end = System.currentTimeMillis();
+            logger.info("String {}", end - start);
+            stringTotalTime += end - start;
+            start = System.currentTimeMillis();
+            tryMatch(inputWrapped, regex);
+            end = System.currentTimeMillis();
+            logger.info("InterruptibleCharSequence {}", end - start);
+            interruptibleCharSequenceTotalTime += end - start;
+        }
+        float overheadPercentage = (float) interruptibleCharSequenceTotalTime / stringTotalTime;
+        logger.info("InterruptibleCharSequence overhead percentage: {}.", overheadPercentage);
+    }
+
+    public boolean tryMatch(CharSequence input, String regex) {
```

Comment:
(optional - could make this method static)
```suggestion
    public static boolean tryMatch(CharSequence input, String regex) {
```